GVRf/Framework/framework/src/main/java/org/gearvrf/GVRAssetLoader.java

```diff
@@ -1014,81 +1019,110 @@ public GVRModelSceneObject loadScene(String filePath, GVRScene scene) throws IOE
     }
 
     /**
-     * Loads a a 3D model and replaces the current scene with it.
-     * The previous scene objects are removed and the loaded model becomes
-     * the only thing in the scene.
+     * Loads a hierarchy of scene objects {@link GVRSceneObject} from a 3D model
+     * replaces the current scene with it.
+     * This function loads the model and its textures asynchronously in the background
+     * and will return before the model is loaded.
+     * IAssetEvents are emitted to event listeners attached to the context.
      *
      * @param model
      *          Scene object to become the root of the loaded model.
      *          This scene object will be named with the base filename of the loaded asset.
-     * @param filePath
-     *            A filename, relative to the root of the volume.
-     *            If the filename starts with "sd:" the file is assumed to reside on the SD Card.
-     *            If the filename starts with "http:" or "https:" it is assumed to be a URL.
-     *            Otherwise the file is assumed to be relative to the "assets" directory.
-     *
+     * @param volume
+     *            A GVRResourceVolume based on the asset path to load.
+     *            This volume will be used as the base for loading textures
+     *            and other models contained within the model.
+     *            You can subclass GVRResourceVolume to provide custom IO.
      * @param scene
      *            Scene to be replaced with the model.
      *
-     * @return A {@link GVRModelSceneObject} that contains the meshes with textures and bones
-     * and animations.
-     * @throws IOException
-     *
      */
-    public GVRSceneObject loadScene(GVRSceneObject model, String filePath, GVRScene scene) throws IOException
+    public void loadScene(final GVRSceneObject model, final GVRResourceVolume volume, final GVRScene scene)
     {
-        AssetRequest assetRequest = new AssetRequest(model, filePath, scene, true);
-        String ext = filePath.substring(filePath.length() - 3).toLowerCase();
+        Threads.spawn(new Runnable()
+        {
+            public void run()
+            {
+                AssetRequest assetRequest = new AssetRequest(model, volume, scene, true);
+                String filePath = volume.getFileName();
+                String ext = filePath.substring(filePath.length() - 3).toLowerCase();
```

Comment:
I think the assumption the file extension is always three characters long may be incorrect. It is something that can wait though.